### PR TITLE
Fixes: a few issues found when running PHP 7.3

### PR DESCRIPTION
**Pull request by josephzidell**
I tried running 2.x on PHP 7.3, and fixed a few simple issues, such as:
* Issue with `preg_match()`
* Undefined variables

### CODE REVIEW

File: lib/Cake/Network/CakeResponse.php [hunk 1]
```diff
@@ -1164,9 +1164,13 @@ public function checkNotModified(CakeRequest $request) {
 		$modifiedSince = $request->header('If-Modified-Since');
 		if ($responseTag = $this->etag()) {
 			$etagMatches = in_array('*', $etags) || in_array($responseTag, $etags);
+		} else {
+			$etagMatches = false;
 		}
 		if ($modifiedSince) {
 			$timeMatches = strtotime($this->modified()) === strtotime($modifiedSince);
+		} else {
+			$timeMatches = false;
```
[the remark this answers is not in the document]
Comment (josephzidell) [reply lib/Cake/Network/CakeResponse.php]: Done

[the remark this answers is not in the document]
Comment (garas) [reply lib/Cake/Network/CakeResponse.php]: But `if (empty($checks))` is now never empty

[the remark this answers is not in the document]
Comment (josephzidell) [reply lib/Cake/Network/CakeResponse.php]: @garas How about we wrap in `array_filter`? `if (empty(array_filter($checks)))`

[the remark this answers is not in the document]
Comment (josephzidell) [reply lib/Cake/Network/CakeResponse.php]: Done

[the remark this answers is not in the document]
Comment (josephzidell) [reply lib/Cake/Network/CakeResponse.php]: Modified. I'll try to get to 3.6 later

File: lib/Cake/Utility/ObjectCollection.php [hunk 2]
```diff
@@ -95,6 +95,7 @@ public function trigger($callback, $params = array(), $options = array()) {
 		if (empty($this->_enabled)) {
 			return true;
 		}
+		$subject = null;
```
Comment (garas) [inline lib/Cake/Utility/ObjectCollection.php]: Tests fail here too.

Very similar problem.

---

To CakePHP Team: Why Travis tests are not running on `2.x`?

**Review by garas — CHANGES_REQUESTED** [review]
Tests fail

File: lib/Cake/Utility/ObjectCollection.php (hunk 2 above)
Comment (markstory) [reply lib/Cake/Utility/ObjectCollection.php]: > To CakePHP Team: Why Travis tests are not running on 2.x?

I'm not sure. Travis used to run tests for 2.x Its possible that there is a problem with the configuration file, or perhaps we're trying to use an unsupported worker type.

Comment (markstory) [reply lib/Cake/Utility/ObjectCollection.php]: It looks like travis ran tests for [2.x 8 days ago](https://travis-ci.org/cakephp/cakephp/builds/416046280) I'm not sure why it isn't running this pull request.

Comment (josephzidell) [reply lib/Cake/Utility/ObjectCollection.php]: Can we merge this as this seems to be a Travis issue?

Comment (garas) [reply lib/Cake/Utility/ObjectCollection.php]: Tests fail here, so should be fixed

**Review by garas — CHANGES_REQUESTED** [review]

File: lib/Cake/Utility/ObjectCollection.php (hunk 2 above)
Comment (josephzidell) [reply lib/Cake/Utility/ObjectCollection.php]: @garas Where can I see the failure?

Comment (garas) [reply lib/Cake/Utility/ObjectCollection.php]: [Run tests](https://book.cakephp.org/2.0/en/development/testing.html#running-tests):

`lib/Cake/Console/cake test core AllTests`

You can also run only some test cases like `Utility/ObjectCollection` or `Network/CakeResponse`

Comment (josephzidell) [reply lib/Cake/Utility/ObjectCollection.php]: I cannot install PHPUnit 3.7, as I already have 7.3 installed on my machine. Can you help me here, by showing me the issue? Could not find it in the PR tools

Comment (garas) [reply lib/Cake/Utility/ObjectCollection.php]: You can install phpunit in CakePHP dir with `composer require --dev phpunit/phpunit:"3.7.38"`

Comment (josephzidell) [reply lib/Cake/Utility/ObjectCollection.php]: Pushed a fix, and now both tests are passing

**Comment by josephzidell** [comment]
Thanks @markstory

**Comment by josephzidell** [comment]
Thanks

**Pull request merged**